docs/manual/transform_data.rst

```diff
@@ -0,0 +1,512 @@
+    1 | 2010-11-23         2     11.5  B_end        11            4   2010           23
+    2 | 2020-02-29         3     12.3  NA            2            9   2020           24.6
+    3 | NA                 4    -13    D_end        NA           16     NA          -26
+   [4 rows x 8 columns]
+
+- Update existing column::
+
+   >>> DT[:, update(strings = f.strings[:1])]
+   >>> DT
+      | dates       integers   floats  strings  months  int_squared   year  float_doubled
+      | date32         int32  float64  str32     int32      float64  int32        float64
+   -- + ----------  --------  -------  -------  ------  -----------  -----  -------------
+    0 | 2000-01-05         1     10    A             1            1   2000           20
+    1 | 2010-11-23         2     11.5  B            11            4   2010           23
+    2 | 2020-02-29         3     12.3  NA            2            9   2020           24.6
+    3 | NA                 4    -13    D            NA           16     NA          -26
+   [4 rows x 8 columns]
+
+
+
+Extend
+^^^^^^
+
+The `extend` method works via :ref:`f-expressions` to creates new columns; it does not update existing columns.
+
+The `extend` method uses a dictionary to create the new columns, where the key in the dictionary is the name of the new column, while the value is the :ref:`f-expressions` to be computed::
+
+   >>> DT = DT[:, :4]
+   >>> DT
+      | dates       integers   floats  strings
+      | date32         int32  float64  str32
+   -- + ----------  --------  -------  -------
+    0 | 2000-01-05         1     10    A
+    1 | 2010-11-23         2     11.5  B
+    2 | 2020-02-29         3     12.3  NA
+    3 | NA                 4    -13    D
+   [4 rows x 4 columns]
+
+- Single column::
+
+   >>> DT = DT[:, f[:].extend({"months" : dt.time.month(f.dates)})]
+   >>> DT
+      | dates       integers   floats  strings  months
+      | date32         int32  float64  str32     int32
+   -- + ----------  --------  -------  -------  ------
+    0 | 2000-01-05         1     10    A             1
+    1 | 2010-11-23         2     11.5  B            11
+    2 | 2020-02-29         3     12.3  NA            2
+    3 | NA                 4    -13    D            NA
+   [4 rows x 5 columns]
+
+- Multiple columns::
+
+   >>> DT = DT[:, f[:].extend({"year" : dt.time.year(f.dates),
+   ...                         "int_squared" : f.integers ** 2})]
+   >>> DT
+      | dates       integers   floats  strings  months   year  int_squared
+      | date32         int32  float64  str32     int32  int32      float64
+   -- + ----------  --------  -------  -------  ------  -----  -----------
+    0 | 2000-01-05         1     10    A             1   2000            1
+    1 | 2010-11-23         2     11.5  B            11   2010            4
+    2 | 2020-02-29         3     12.3  NA            2   2020            9
+    3 | NA                 4    -13    D            NA     NA           16
+   [4 rows x 7 columns]
```

Comment:
I would add a note here emphasizing that in contrast to `update` we need to assign the result to `DT`.